backend/plonk/bn254/solidity.go

```diff
@@ -422,7 +421,6 @@ contract PlonkVerifier {
   }
 
   function compute_pi(
```

Comment:
You've lost some of the nice comments you wrote for the audit (e.g. for `compute_pi()` but also in other places).  Maybe you should copy these over if this is the master copy of the contract?

Reply:
Ah nice catch, the last test that I generated was without the use of api.Commit() and the proof is not needed in that case I forgot about that (I followed the compiler that told me the parameter was unused). I'll add a condition in the template to not take the proof when commit is not used

Reply:
I re added the comments